src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1030,6 +1030,7 @@ public function withCount($relations)
                 $relation->getRelated()->newQuery(), $this
             );
 
+            $this->mergeModelDefinedRelationWheresToHasQuery($query, $relation);
             call_user_func($constraints, $query);
```

Comment:
So something like this? https://github.com/laravel/framework/pull/13612/commits/4abb185e5a60148676df2333fe36ed9c31f8f480